src/Microsoft.DotNet.GenFacades/NotSupportedAssemblyGenerator.cs

```diff
@@ -28,11 +28,14 @@ public class NotSupportedAssemblyGenerator : BuildTask
 
         public override bool Execute()
         {
-           GenerateNotSupportedAssemblyFiles(SourceFiles?.Select(item => item.ItemSpec).ToList(), IntermediateOutputPath, Message, ApiExclusionListPath);
-           return !Log.HasLoggedErrors;
+            if (SourceFiles == null || SourceFiles.Length == 0)
+                Log.LogError("There are no ref source files.");
+
+            GenerateNotSupportedAssemblyFiles(SourceFiles?.Select(item => item.ItemSpec).ToList(), IntermediateOutputPath, Message, ApiExclusionListPath);
```

Comment:
```suggestion
            if (SourceFiles == null || SourceFiles.Length == 0)
            {
                Log.LogError("There are no ref source files.");
            }
           else
           {
                GenerateNotSupportedAssemblyFiles(SourceFiles.Select(item => item.ItemSpec).ToList(), IntermediateOutputPath, Message, ApiExclusionListPath);   
            }
```

Log.LogError doesn't exit so we need to either exit or use if else.